Added "No Silver Bullet"

tallesl opened the pull request:
I was surprised that this classic paper wasn't submitted already.
"No Silver Bullet" (along with the great "Out of the Tar Pit") is one of my favourites papers.

I love how he tackled the difficulties and the silver bullets one by one through the paper.
Ending with an ought to design shows how this paper aged so well (in todays world where software development is often just "coding").

zeeshanlakhani commented:
@tallesl I'm +1 to this, but I've made a comment on the diff that needs changing and a rebase.

zeeshanlakhani commented on design/README.md:
```diff
@@ -1,6 +1,9 @@
-* :scroll: [Out of the Tar Pit](out-of-the-tar-pit.pdf)
-    > We identify common causes of complexity and discuss general approaches which can be taken to eliminate them where they are accidental in nature.
\ No newline at end of file
+* [Out of the Tar Pit (2006)](out-of-the-tar-pit.pdf)
```
We want to keep the :scroll: on this, as we have the paper hosted within the directory.

zeeshanlakhani commented:
@tallesl Mind updating based on my comment?

tallesl commented:
Oh, I'm so sorry about that, I didn't know (maybe stating on [contributing](https://github.com/papers-we-love/papers-we-love#contributing-guidelines) is a good idea?).

I fixed (and rebased with the current master).

zeeshanlakhani commented:
Yep, I'll update the contribution section too. Thanks!

DarrenN commented:
@zeeshanlakhani / @tallesl can we get this up to date with master and merged?

DarrenN commented:
Closed due to inactivity. Please re-open once the outstanding issues are addressed.
